### PR TITLE
fix(theme): pin the default palette's text companions per appearance

### PR DESCRIPTION
**Pull request by fx**
## Summary

The Default theme pinned all ten `-text` companions to their scale's Radix step 11. Step 11 is calibrated against steps 1–2 of its *own* scale rather than against a neutral, so how it measures on Liebe's card surface depends on which appearance is active — and on the light card three of the ten fall under the 4.5:1 text floor. This pins those three to step 12 in light appearance only, and restores step 11 in a dark block.

Completes **PR 1 only** of `docs/changes/0035-light-appearance-contrast.md`. PR 2 (glyph on its 20% tint) and PR 3 (residual accessible names) are untouched, so **the change document stays `draft`** — `docs/index.yml` and `docs/index.md` are deliberately not modified. Supersedes issue #197.

## What changed

`src/theme/themes/default.css` now mirrors how `src/styles/tokens.css` handles the surface tokens: Radix's classless default is light, so the light values are unconditional and dark overrides them. The dark block contains nothing but the three step-11 restorations.

| Triplet | Light | Dark |
| --- | --- | --- |
| `--liebe-c-light-text` | `amber-12` (was `amber-11`) | `amber-11` |
| `--liebe-c-heat-text` | `orange-12` (was `orange-11`) | `orange-11` |
| `--liebe-c-vacuum-text` | `teal-12` (was `teal-11`) | `teal-11` |
| the other seven | `<scale>-11`, unchanged | `<scale>-11`, unchanged |

The other seven stay on step 11 in light. The change document requires this explicitly: pulling hues that already clear the floor darker would change the state line's weight on every card of those domains to fix a defect three hues have.

## Measured figures

Method: render Radix's `styles.css` plus the base token layer and the theme layer in Chromium, under the same root attributes `LiebeThemeProvider` produces (`data-accent-color="indigo"`, `data-gray-color="slate"` — the auto-gray pairing), then screenshot and decode the painted pixels and compute the WCAG 2.1 ratio from the decoded sRGB triples. Nothing is read off a token value in a file, so the `var()` chain, the auto-gray mapping and the appearance selectors are all exercised. The rig reproduces the three figures the change document recorded to the digit (4.497 / 4.398 / 4.450), which is what qualifies it as a measurement of the same thing.

#### Light appearance — card surface `--liebe-card-bg` → `--gray-1`, #fcfcfd

| Triplet | Before | Ratio | After | Ratio | 4.5:1 floor |
| --- | --- | --- | --- | --- | --- |
| `--liebe-c-light-text` | amber-11 `#ab6400` | 4.497 | amber-12 `#4f3422` | 11.091 | pass (was **fail**) |
| `--liebe-c-heat-text` | orange-11 `#cc4e00` | 4.398 | orange-12 `#582d1d` | 11.328 | pass (was **fail**) |
| `--liebe-c-cool-text` | sky-11 `#00749e` | 5.131 | _unchanged_ | 5.131 | pass |
| `--liebe-c-ok-text` | green-11 `#218358` | 4.600 | _unchanged_ | 4.600 | pass |
| `--liebe-c-alert-text` | red-11 `#ce2c31` | 5.084 | _unchanged_ | 5.084 | pass |
| `--liebe-c-media-text` | indigo-11 `#3a5bc7` | 5.850 | _unchanged_ | 5.850 | pass |
| `--liebe-c-vacuum-text` | teal-11 `#008573` | 4.450 | teal-12 `#0d3d38` | 11.759 | pass (was **fail**) |
| `--liebe-c-water-text` | cyan-11 `#107d98` | 4.646 | _unchanged_ | 4.646 | pass |
| `--liebe-c-default-text` | blue-11 `#0d74ce` | 4.648 | _unchanged_ | 4.648 | pass |
| `--liebe-c-brand-text` | crimson-11 `#cb1d63` | 5.258 | _unchanged_ | 5.258 | pass |

#### Dark appearance — card surface `--liebe-card-bg` → `--gray-2`, #18191b

| Triplet | Before | Ratio | After | Ratio | 4.5:1 floor |
| --- | --- | --- | --- | --- | --- |
| `--liebe-c-light-text` | amber-11 `#ffca16` | 11.487 | _unchanged_ | 11.487 | pass |
| `--liebe-c-heat-text` | orange-11 `#ffa057` | 8.727 | _unchanged_ | 8.727 | pass |
| `--liebe-c-cool-text` | sky-11 `#75c7f0` | 9.369 | _unchanged_ | 9.369 | pass |
| `--liebe-c-ok-text` | green-11 `#3dd68c` | 9.379 | _unchanged_ | 9.379 | pass |
| `--liebe-c-alert-text` | red-11 `#ff9592` | 8.349 | _unchanged_ | 8.349 | pass |
| `--liebe-c-media-text` | indigo-11 `#9eb1ff` | 8.534 | _unchanged_ | 8.534 | pass |
| `--liebe-c-vacuum-text` | teal-11 `#0bd8b6` | 9.644 | _unchanged_ | 9.644 | pass |
| `--liebe-c-water-text` | cyan-11 `#4ccce6` | 9.288 | _unchanged_ | 9.288 | pass |
| `--liebe-c-default-text` | blue-11 `#70b8ff` | 8.368 | _unchanged_ | 8.368 | pass |
| `--liebe-c-brand-text` | crimson-11 `#ff92ad` | 8.350 | _unchanged_ | 8.350 | pass |

Worst light-appearance figure goes from 4.398:1 (failing) to 4.600:1 (`--liebe-c-ok-text`, unchanged and already passing). **Every dark figure is byte-identical before and after** — verified by diffing the two measurement runs, which reports exactly three changed rows, all in light. Dark appearance is out of scope per the change document, and a PR that moved a dark figure would have overreached.

For reference, the other two built-in themes were measured with the same rig and need no change: LCARS clears the floor on all ten in both appearances (5.05:1 worst, its own palette on `--lcars-black`), and Liquid Glass clears it on all ten (9.70:1 worst) since it already pins step 12 throughout — its `--liebe-c-default-text` measures 11.397:1 against the 11.37:1 the change document cites.

## Spec updates

Spec restatements land in the same PR as the behaviour change, per the rule at the top of the change document's task list.

- `docs/specs/design-system/index.md` — the domain-colour bullet no longer says Default pins step 11 flat. It now states that a `-text` pin MUST clear the floor **in each appearance separately** (with the reason: step 11 is calibrated against its own scale, so clearing one appearance is no evidence for the other), names the three light-appearance exceptions, and records that darkening a passing hue is not part of the remedy. The outstanding-defect list drops the text-AA entry — five open defects become four, with the closed one recorded rather than deleted so the history stays checkable. Changelog row added.
- `docs/specs/theming/index.md` — the Default-theme caveat stopped naming a Radix step of its own (that is the design system's to own) and now tells user CSS replacing a pinned `-text` to cover both appearances. Changelog row added.
- `src/components/GridCard.stories.tsx` — the `ColorOverride` story told a reviewer to *expect* `color-contrast` on the `light`, `heat` and `vacuum` state lines as a known defect. That note now describes something that no longer happens, and expected-failure guidance outliving its defect is how a real regression gets waved through as already-tracked; it now records the reverse.

## Test changes

`src/theme/__tests__/tokens.test.ts` asserted `-text` from a **whole-file** declaration map. With a per-appearance sheet that map returns the dark value for any token both blocks declare, so it would have kept passing while silently no longer checking the light value — the appearance the defect was in. It now splits the sheet on the dark selector and asserts each block:

- every `-text` in the light block resolves to the step that measures on the light card;
- the dark block's **full key list** is exactly the three restorations, so an unasked-for dark-appearance change fails rather than passing quietly;
- the tints stay derived in both blocks.

### Probes

Committed first, then mutated, each mutation confirmed applied via `git diff --quiet` before reading the result, and confirmed restored after. The tests read the stylesheet from disk with `readFileSync`, so there is no bundler, dev server or `dist/` between the mutation and what executes — the artifact under test is the mutated file itself.

| Mutation | Caught by |
| --- | --- |
| light `--liebe-c-light-text` back to `amber-11` | `pins every text companion to the step that measures on the light card` |
| extra `--liebe-c-ok-text` added to the dark block | `restores step 11 in dark for exactly the hues light darkens` |
| dark block's selector deleted | `opens its dark block with the same selector the base sheet uses` |
| dark `--liebe-c-heat-text` moved to `orange-12` | `restores step 11 in dark for exactly the hues light darkens` |

I also removed one assertion I had written that compared a literal to a literal (`expect([...LIGHT_TEXT_STEPS.keys()]).toEqual([...])`) — it could not fail for any change to the stylesheet, and a test that pins nothing reads like a guard.

## Gates

- `npm test` — **pass**, 226 files / 4891 tests.
- `npm run lint` — **pass**, 0 errors and the one pre-existing `MigrationCutoffs` warning in `src/store/persistence.ts`.
- `npm run typecheck` — **pass**.
- `npm run test:coverage` — the patch has **no coverable lines**: the only source-tree changes are a `.css` file (not in the v8 JS denominator) and files under `__tests__/` and `*.stories.tsx`, both excluded by `vitest.config.ts`. The behaviour is still gated, by the token contract test above.

One thing to flag rather than bury: the full instrumented coverage run failed on config-modal and icon-picker specs, **8 failures on one run and a different 4 on another, all `Test timed out in 5000ms`** and none of them touching theme tokens. They pass in isolation under `--coverage`. That is load-sensitive flakiness in the harness, not this change — the same class of problem `docs/changes/0040-test-harness-reliability.md` exists for — but it means the local coverage run is not currently a clean signal for anyone.

## Reviews

- `codex review --base main` — ran; it caught the stale `ColorOverride` expected-failure comment described above, which is fixed in this PR.
- CodeRabbit (`cr --base main --agent`) — **skipped (rate-limited)**: `Rate limit exceeded`, 10 minute cooldown.

## Not done

Nothing in PR 1's scope. Worth naming for whoever picks up PR 2: in light appearance the three exception hues now render their state line noticeably darker than the other seven, since step 12 is a near-desaturated dark of the hue. That asymmetry is the specified remedy — the change document rejects both a blanket step-12 repin and accepting 4.4:1 — but it is a visible consequence, and if it reads badly the decision to revisit is the change document's, not mine.

### CODE REVIEW

File: docs/specs/design-system/index.md [hunk 1]
```diff
@@ -102,8 +102,8 @@ These MUST be declared on the themed root and inherited, not applied per compone
...
...hing catches it: the derivation is what keeps a base-only remap coherent, so there is no invalid value to detect and no gate that fails. This is why Liquid Glass pins step 12 throughout and Default pins step 11 rather than inheriting the derivation. **A pin MUST clear the 4.5:1 text floor against the card surface in each appearance separately**, because a Radix step 11 is calibrated against steps 1–2 of its own scale rather than against a neutral: the same step measures differently on the light card surface than on the dark one, so clearing the floor in one appearance is no evidence for the other. Default consequently pins step 12 in **light** appearance for the three hues whose step 11 measures short there — `--liebe-c-light` (amber), `--liebe-c-heat` (orange), `--liebe-c-vacuum` (teal) — and step 11 for the other seven in light and for all ten in dark; darkening a hue that already clears the floor is a visual change to every card of that domain, not part of the remedy. The glyph-on-tint pairing the same triplet drives is a separate calibration and still misses 3:1 in light ([0035](../../changes/0035-light-appearance-contrast.md)). Pinning is the theme's job, not the base layer's.
```
[the remark this answers is not in the document]
Comment (fx) [reply docs/specs/design-system/index.md]: Valid — fixed in 53e8b90. The clause now reads "Liquid Glass pins step 12 throughout and Default pins `-text` explicitly per appearance rather than inheriting the derivation". You were right that this was more than a stale phrasing: the very next sentence in the same bullet already stated the per-appearance rule correctly, so the paragraph was contradicting itself, and a reader skimming the first clause took away a flat step-11 pin that the code does not do.

File: src/theme/__tests__/tokens.test.ts [hunk 2]
```diff
@@ -54,7 +54,41 @@ function cssImports(source: string): string[] {
...
 const themeRules = stripComments(defaultThemeCss)
 const base = declarations(baseRules)
-const theme = declarations(themeRules)
+
+/**
+ * The selector both sheets open their dark-appearance block with. Radix's
+ * classless default is light, so light is the unconditional value and dark is
+ * the override.
+ */
+const DARK_SELECTOR = /\.radix-themes:where\(\.dark, \.dark-theme\),/
```
[the remark this answers is not in the document]
Comment (fx) [reply src/theme/__tests__/tokens.test.ts]: Valid — fixed in 53e8b90. `DARK_SELECTOR` is now `/\.radix-themes:where\(\s*\.dark\s*,\s*\.dark-theme\s*\)\s*,/`: whitespace optional everywhere CSS permits it, but still rejected between `.radix-themes` and `:where(`, where whitespace would make it a descendant combinator and change the selector's meaning.

Loosening a pattern risks matching anything, so it was probed against all four shapes rather than one: a cosmetic reformat to `.dark,.dark-theme ,` in both sheets is now tolerated (green, the intent of the change), while dropping `.dark-theme` from the list, renaming it to `.dark-mode`, and deleting the dark selector line entirely each turn the suite red. That last case is the one that mattered — an over-loosened pattern would have passed with the dark block gone.
